src/clj_icat_direct/queries.clj

```diff
@@ -565,6 +565,14 @@
                 AND c.parent_coll_name = ANY(ARRAY( SELECT coll_name FROM parent ))
                 AND c.coll_type != 'linkPoint')) AS total"
 
+   :list-files-under-folder
+   "SELECT c.coll_name || '/' || d.data_name AS path
+      FROM r_coll_main c
+      JOIN r_data_main d ON c.coll_id = d.coll_id
+     WHERE c.coll_name = ?
+        OR c.coll_name LIKE ? || '/%'
+  ORDER BY c.coll_name, d.data_name"
```

Comment:
Loathe as I am to impose such a thing on you, you probably will want to handle the `data_repl_num` stuff here. See for example `mk-unique-objs-in-coll` in this same file for an example of what I mean. For your use cases, it might be enough to add a `DISTINCT ON (d.data_id)`, possibly.

Reply:
I suppose, thinking about it more, that a plain old `DISTINCT` (or throwing out non-unique values higher up, but probably better to do it in the DB) might also work too.

Reply:
Ah, thanks for catching that. I'd added `DISTINCT` to my test query and forgotten to add it to the source.